Using make directly means submakes can't be parallell, so use $(MAKE).

Upstream-Status: Submitted [https://github.com/projectacrn/acrn-hypervisor/pull/2360]
Signed-off-by: Ross Burton <ross.burton@intel.com>

From e536077615875f2ec6ab560071d1e5d2a5a623f3 Mon Sep 17 00:00:00 2001
From: Ross Burton <ross.burton@intel.com>
Date: Wed, 16 Jan 2019 21:35:20 +0000
Subject: [PATCH] Use $(MAKE)

---
 Makefile             | 34 +++++++++++++++++-----------------
 devicemodel/Makefile |  4 ++--
 tools/Makefile       | 28 ++++++++++++++--------------
 3 files changed, 33 insertions(+), 33 deletions(-)

diff --git a/Makefile b/Makefile
index 86580067..8bde7139 100644
--- a/Makefile
+++ b/Makefile
@@ -46,33 +46,33 @@ export TOOLS_OUT
 all: hypervisor devicemodel tools
 
 hypervisor:
-	make -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT) BOARD=$(BOARD) FIRMWARE=$(FIRMWARE) RELEASE=$(RELEASE) clean
-	make -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT) BOARD=$(BOARD) FIRMWARE=$(FIRMWARE) RELEASE=$(RELEASE)
+	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT) BOARD=$(BOARD) FIRMWARE=$(FIRMWARE) RELEASE=$(RELEASE) clean
+	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT) BOARD=$(BOARD) FIRMWARE=$(FIRMWARE) RELEASE=$(RELEASE)
 ifeq ($(FIRMWARE),uefi)
 	echo "building hypervisor as EFI executable..."
-	make -C $(T)/efi-stub HV_OBJDIR=$(HV_OUT) EFI_OBJDIR=$(EFI_OUT)
+	$(MAKE) -C $(T)/efi-stub HV_OBJDIR=$(HV_OUT) EFI_OBJDIR=$(EFI_OUT)
 endif
 
 sbl-hypervisor:
 	@mkdir -p $(HV_OUT)-sbl
-	make -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-sbl BOARD=apl-mrb FIRMWARE=sbl RELEASE=$(RELEASE) clean
-	make -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-sbl BOARD=apl-mrb FIRMWARE=sbl RELEASE=$(RELEASE)
+	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-sbl BOARD=apl-mrb FIRMWARE=sbl RELEASE=$(RELEASE) clean
+	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-sbl BOARD=apl-mrb FIRMWARE=sbl RELEASE=$(RELEASE)
 
 devicemodel: tools
-	make -C $(T)/devicemodel DM_OBJDIR=$(DM_OUT) clean
-	make -C $(T)/devicemodel DM_OBJDIR=$(DM_OUT) DM_BUILD_VERSION=$(BUILD_VERSION) DM_BUILD_TAG=$(BUILD_TAG)
+	$(MAKE) -C $(T)/devicemodel DM_OBJDIR=$(DM_OUT) clean
+	$(MAKE) -C $(T)/devicemodel DM_OBJDIR=$(DM_OUT) DM_BUILD_VERSION=$(BUILD_VERSION) DM_BUILD_TAG=$(BUILD_TAG)
 
 tools:
 	mkdir -p $(TOOLS_OUT)
-	make -C $(T)/tools OUT_DIR=$(TOOLS_OUT) RELEASE=$(RELEASE)
+	$(MAKE) -C $(T)/tools OUT_DIR=$(TOOLS_OUT) RELEASE=$(RELEASE)
 
 doc:
-	make -C $(T)/doc html BUILDDIR=$(DOC_OUT)
+	$(MAKE) -C $(T)/doc html BUILDDIR=$(DOC_OUT)
 
 .PHONY: clean
 clean:
-	make -C $(T)/tools OUT_DIR=$(TOOLS_OUT) clean
-	make -C $(T)/doc BUILDDIR=$(DOC_OUT) clean
+	$(MAKE) -C $(T)/tools OUT_DIR=$(TOOLS_OUT) clean
+	$(MAKE) -C $(T)/doc BUILDDIR=$(DOC_OUT) clean
 	rm -rf $(ROOT_OUT)
 
 .PHONY: install
@@ -80,18 +80,18 @@ install: hypervisor-install devicemodel-install tools-install
 
 hypervisor-install:
 ifeq ($(FIRMWARE),sbl)
-	make -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT) BOARD=$(BOARD) FIRMWARE=$(FIRMWARE) RELEASE=$(RELEASE) install
+	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT) BOARD=$(BOARD) FIRMWARE=$(FIRMWARE) RELEASE=$(RELEASE) install
 endif
 ifeq ($(FIRMWARE),uefi)
-	make -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT) BOARD=$(BOARD) FIRMWARE=$(FIRMWARE) RELEASE=$(RELEASE)
-	make -C $(T)/efi-stub HV_OBJDIR=$(HV_OUT) EFI_OBJDIR=$(EFI_OUT) all install
+	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT) BOARD=$(BOARD) FIRMWARE=$(FIRMWARE) RELEASE=$(RELEASE)
+	$(MAKE) -C $(T)/efi-stub HV_OBJDIR=$(HV_OUT) EFI_OBJDIR=$(EFI_OUT) all install
 endif
 
 sbl-hypervisor-install:
-	make -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-sbl BOARD=apl-mrb FIRMWARE=sbl RELEASE=$(RELEASE) install
+	$(MAKE) -C $(T)/hypervisor HV_OBJDIR=$(HV_OUT)-sbl BOARD=apl-mrb FIRMWARE=sbl RELEASE=$(RELEASE) install
 
 devicemodel-install:
-	make -C $(T)/devicemodel DM_OBJDIR=$(DM_OUT) install
+	$(MAKE) -C $(T)/devicemodel DM_OBJDIR=$(DM_OUT) install
 
 tools-install:
-	make -C $(T)/tools OUT_DIR=$(TOOLS_OUT) RELEASE=$(RELEASE) install
+	$(MAKE) -C $(T)/tools OUT_DIR=$(TOOLS_OUT) RELEASE=$(RELEASE) install
diff --git a/devicemodel/Makefile b/devicemodel/Makefile
index 5cb052f9..51a91ed7 100644
--- a/devicemodel/Makefile
+++ b/devicemodel/Makefile
@@ -166,7 +166,7 @@ all: $(DM_OBJDIR)/$(PROGRAM)
 	@echo -n ""
 
 $(VMCFG_CONFIG_H):
-	make -C $(BASEDIR)/vmcfg $@ BASEDIR=$(BASEDIR) DM_OBJDIR=$(DM_OBJDIR)
+	$(MAKE) -C $(BASEDIR)/vmcfg $@ BASEDIR=$(BASEDIR) DM_OBJDIR=$(DM_OBJDIR)
 
 $(DM_OBJDIR)/$(PROGRAM): $(OBJS)
 	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(LIBS)
@@ -229,4 +229,4 @@ install-bios: $(BIOS_BIN)
 	install -D --mode=0664 -t $(DESTDIR)$(datadir)/acrn/bios $^
 
 install-vmcfg:
-	make -C $(BASEDIR)/vmcfg install DESTDIR=$(DESTDIR) BASEDIR=$(BASEDIR)
+	$(MAKE) -C $(BASEDIR)/vmcfg install DESTDIR=$(DESTDIR) BASEDIR=$(BASEDIR)
diff --git a/tools/Makefile b/tools/Makefile
index ed5077e8..f0102427 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -9,26 +9,26 @@ all: acrnlog acrn-manager acrntrace acrnbridge
 endif
 
 acrn-crashlog:
-	make -C $(T)/acrn-crashlog OUT_DIR=$(OUT_DIR) RELEASE=$(RELEASE)
+	$(MAKE) -C $(T)/acrn-crashlog OUT_DIR=$(OUT_DIR) RELEASE=$(RELEASE)
 
 acrnlog:
-	make -C $(T)/acrnlog OUT_DIR=$(OUT_DIR)
+	$(MAKE) -C $(T)/acrnlog OUT_DIR=$(OUT_DIR)
 
 acrn-manager:
-	make -C $(T)/acrn-manager OUT_DIR=$(OUT_DIR) RELEASE=$(RELEASE)
+	$(MAKE) -C $(T)/acrn-manager OUT_DIR=$(OUT_DIR) RELEASE=$(RELEASE)
 
 acrntrace:
-	make -C $(T)/acrntrace OUT_DIR=$(OUT_DIR)
+	$(MAKE) -C $(T)/acrntrace OUT_DIR=$(OUT_DIR)
 
 acrnbridge:
-	make -C $(T)/acrnbridge OUT_DIR=$(OUT_DIR)
+	$(MAKE) -C $(T)/acrnbridge OUT_DIR=$(OUT_DIR)
 
 .PHONY: clean
 clean:
-	make -C $(T)/acrn-crashlog OUT_DIR=$(OUT_DIR) clean
-	make -C $(T)/acrn-manager OUT_DIR=$(OUT_DIR) clean
-	make -C $(T)/acrntrace OUT_DIR=$(OUT_DIR) clean
-	make -C $(T)/acrnlog OUT_DIR=$(OUT_DIR) clean
+	$(MAKE) -C $(T)/acrn-crashlog OUT_DIR=$(OUT_DIR) clean
+	$(MAKE) -C $(T)/acrn-manager OUT_DIR=$(OUT_DIR) clean
+	$(MAKE) -C $(T)/acrntrace OUT_DIR=$(OUT_DIR) clean
+	$(MAKE) -C $(T)/acrnlog OUT_DIR=$(OUT_DIR) clean
 	rm -rf $(OUT_DIR)
 
 .PHONY: install
@@ -39,16 +39,16 @@ install: acrnlog-install acrn-manager-install acrntrace-install acrnbridge-insta
 endif
 
 acrn-crashlog-install:
-	make -C $(T)/acrn-crashlog OUT_DIR=$(OUT_DIR) install
+	$(MAKE) -C $(T)/acrn-crashlog OUT_DIR=$(OUT_DIR) install
 
 acrnlog-install:
-	make -C $(T)/acrnlog OUT_DIR=$(OUT_DIR) install
+	$(MAKE) -C $(T)/acrnlog OUT_DIR=$(OUT_DIR) install
 
 acrn-manager-install:
-	make -C $(T)/acrn-manager OUT_DIR=$(OUT_DIR) install
+	$(MAKE) -C $(T)/acrn-manager OUT_DIR=$(OUT_DIR) install
 
 acrntrace-install:
-	make -C $(T)/acrntrace OUT_DIR=$(OUT_DIR) install
+	$(MAKE) -C $(T)/acrntrace OUT_DIR=$(OUT_DIR) install
 
 acrnbridge-install:
-	make -C $(T)/acrnbridge OUT_DIR=$(OUT_DIR) install
+	$(MAKE) -C $(T)/acrnbridge OUT_DIR=$(OUT_DIR) install
-- 
2.11.0

